File: LeetCode/1071/1071.py
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        if str1 + str2 != str2 + str1:
            return ""
        def gcd(str1, str2):
            minval = min(len(str1), len(str2))
            for i in range(minval, 0, -1):
                if len(str1) % i == 0 and len(str2) % i ==0:
                    return i
            return 1
        return str1[:gcd(str1, str2)]

File: LeetCode/1071/1071.cpp
class Solution {
public:
    string gcdOfStrings(string str1, string str2) {
        
        if(str1 + str2 != str2 + str1){
            return "";
        }
        int result = gcd(str1.length(), str2.length());
        return str1.substr(0, result);
        
    }
private:
    int gcd(int a, int b){
        while (b != 0){
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    };
};
